src/hotspot/share/opto/macro.cpp

```diff
@@ -573,6 +573,9 @@ bool PhaseMacroExpand::can_eliminate_allocation(PhaseIterGVN* igvn, AllocateNode
     if (res_type == nullptr) {
       NOT_PRODUCT(fail_eliminate = "Neither instance or array allocation";)
       can_eliminate = false;
+    } else if (!res_type->klass_is_exact()) {
+      NOT_PRODUCT(fail_eliminate = "Not an exact type.";)
+      can_eliminate = false;
```

Comment:
You already fixed this: [#18851](https://github.com/openjdk/jdk/pull/18851)
Please, merge latest changes into this PR.

Also you need new regression tests.